src/root/async.c

```diff
@@ -109,7 +114,8 @@ unsigned __stdcall startthread(void *p)
     AsyncRead *aw = (AsyncRead *)p;
 
     //printf("aw->filesdim = %p %d\n", aw, aw->filesdim);
-    for (size_t i = 0; i < aw->filesdim; i++)
+    size_t dim = aw->filesdim;
+    for (size_t i = 0; i < dim; i++)
```

Comment:
Now that `AsyncRead::dispose()` waits for all operations on all files to complete, it should not be possible to `free()` `aw`, before `startthread()` completes.
Even if this somehow happens, then accessing `aw->filesdim` or using a cached version of it is irrelevant, because inside the loop we access the `aw->files` member, which would also lead to the same problem.